### ISSUE TITLE
GHC 8.4.1 build failure

### ISSUE THREAD

**Issue by ilovezfs** [solidsnack/arx#7]
```
bash-3.2$ cabal install --jobs=8 --max-backjumps=100000 --prefix=/usr/local/Cellar/arx/0.2.3
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Resolving dependencies...
Notice: installing into a sandbox located at
/private/tmp/arx-20180322-34085-wwhcwq/arx-0.2.3/.cabal-sandbox
Configuring arx-0.2.3...
Building arx-0.2.3...
Failed to install arx-0.2.3
Build log ( /private/tmp/arx-20180322-34085-wwhcwq/arx-0.2.3/.cabal-sandbox/logs/ghc-8.4.1/arx-0.2.3-KZlIEVs7LlkHdOpLORldu2.log ):
cabal: Entering directory '.'
Configuring arx-0.2.3...
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Preprocessing executable 'arx' for arx-0.2.3..
Building executable 'arx' for arx-0.2.3..

<no location info>: warning: [-Wmissing-home-modules]
    These modules are needed for compilation but not listed in your .cabal file's other-modules: System.Posix.ARX
                                                                                                 System.Posix.ARX.CLI
                                                                                                 System.Posix.ARX.CLI.CLTokens
                                                                                                 System.Posix.ARX.CLI.Options
                                                                                                 System.Posix.ARX.HEREDat
                                                                                                 System.Posix.ARX.Programs
                                                                                                 System.Posix.ARX.Sh
                                                                                                 System.Posix.ARX.TMPXTools
                                                                                                 System.Posix.ARX.Tar
[ 1 of 10] Compiling System.Posix.ARX.CLI.CLTokens ( System/Posix/ARX/CLI/CLTokens.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/CLI/CLTokens.o )
[ 2 of 10] Compiling System.Posix.ARX.HEREDat ( System/Posix/ARX/HEREDat.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/HEREDat.o )
[ 3 of 10] Compiling System.Posix.ARX.Sh ( System/Posix/ARX/Sh.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/Sh.o )
[ 4 of 10] Compiling System.Posix.ARX.CLI.Options ( System/Posix/ARX/CLI/Options.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/CLI/Options.o )
[ 5 of 10] Compiling System.Posix.ARX.TMPXTools ( System/Posix/ARX/TMPXTools.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/TMPXTools.o )
[ 6 of 10] Compiling System.Posix.ARX.Tar ( System/Posix/ARX/Tar.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/Tar.o )
[ 7 of 10] Compiling System.Posix.ARX.Programs ( System/Posix/ARX/Programs.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/Programs.o )
[ 8 of 10] Compiling System.Posix.ARX ( System/Posix/ARX.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX.o )
[ 9 of 10] Compiling System.Posix.ARX.CLI ( System/Posix/ARX/CLI.hs, dist/dist-sandbox-41c3296e/build/arx/arx-tmp/System/Posix/ARX/CLI.o )

System/Posix/ARX/CLI.hs:153:10: error:
    • No instance for (Semigroup ZOM)
        arising from the superclasses of an instance declaration
    • In the instance declaration for ‘Monoid ZOM’
    |
153 | instance Monoid ZOM where
    |          ^^^^^^^^^^
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
arx-0.2.3-KZlIEVs7LlkHdOpLORldu2 failed during the building phase. The
exception was:
ExitFailure 1
bash-3.2$ 
```

**Comment by solidsnack** [comment]
Fixed in: 8ec85a7

**Comment by ilovezfs** [comment]
Thanks, @solidsnack! Voilà https://github.com/Homebrew/homebrew-core/pull/25753
